### PR TITLE
fix: checking out a non-existing branch fails the project sync

### PR DESCRIPTION
**Pull request by bzwei**
Need to distinguish a git checkout failure is due to a non-existing branch/tag/commit, or due to an empty repository. The raw error messages are very similar but we have set the project sync state to completed for the later case in an earlier fix. This fix identifies the first case and set the state to be failed.

AAP-29144: Project sync should report status as "failed" when using non-existent branch

Also refactor the existing code to use a proper error class instead of. an error message to identify the case for an empty repo.

Will add e2e tests for the two cases.

### CODE REVIEW

**Comment by hsong-rh** [comment]
@bzwei You will add pytest for the change, right?

**Comment by bzwei** [comment]
> @bzwei You will add pytest for the change, right?

yes, new tests will be added to eda_qa

File: src/aap_eda/services/project/imports.py [hunk 1]
```diff
@@ -61,13 +61,12 @@ def wrapper(self: ProjectImportService, project: models.Project):
...
-        except Exception as e:
+        except ScmEmptyError as e:
             # if a project is empty, sync status should show completed
-            project.import_state = (
-                models.Project.ImportState.COMPLETED
-                if "Project folder is empty" in e.args[0]
-                else models.Project.ImportState.FAILED
-            )
+            project.import_state = models.Project.ImportState.COMPLETED
+            project.import_error = str(e)
```
Comment (Alex-Izquierdo) [inline src/aap_eda/services/project/imports.py:67]: It doesn't make sense to set an "import error" if the state is not error. Here I see a potential inconsistency.

File: src/aap_eda/services/project/imports.py [hunk 2]
```diff
@@ -61,13 +61,12 @@ def wrapper(self: ProjectImportService, project: models.Project):
...
             project.import_state = models.Project.ImportState.COMPLETED
-        except Exception as e:
+        except ScmEmptyError as e:
             # if a project is empty, sync status should show completed
-            project.import_state = (
-                models.Project.ImportState.COMPLETED
-                if "Project folder is empty" in e.args[0]
-                else models.Project.ImportState.FAILED
-            )
+            project.import_state = models.Project.ImportState.COMPLETED
```
Comment (Alex-Izquierdo) [inline src/aap_eda/services/project/imports.py:66]: should be this failed? in the description you say: 
> This fix identifies the first case and set the state to be failed.

File: src/aap_eda/services/project/imports.py (hunk 1 above)
Comment (bzwei) [reply src/aap_eda/services/project/imports.py:67]: As discussed in the SU. This field should be renamed in another issue when it is time to change UI together.

File: src/aap_eda/services/project/imports.py (hunk 2 above)
Comment (bzwei) [reply src/aap_eda/services/project/imports.py:66]: ScmEmptyError is for the second case where the repo is truly empty. We want it to be completed.

**Review by Alex-Izquierdo — APPROVED** [review]

**Review by zkayyali812 — APPROVED** [review]

**Pull request merged**